lib/src/view_model.dart

```diff
@@ -52,6 +83,7 @@ abstract class ViewModel<State, Effect> {
     }
   }
 
+  /// Closes the [ViewModel]
```

Comment:
/// Closes the ViewModel, completing associated controllers.
///
/// It closes the state and effect controllers, marking the ViewModel as closed.
///
/// During the close operation, the state controller and effect controller are closed using
/// asynchronous operations, and the `isClosed` flag is set to `true`.
///
/// Subclasses should call `super.close()` as part of their overridden `close` methods.
///
/// Throws:
///   - Any error that occurs during the closing of controllers is propagated.